src/sources.cpp

```diff
@@ -279,6 +280,7 @@ static void src_vol_chunkloop(fields_chunk *fc, int ichunk, component c,
   LOOP_OVER_DIRECTIONS(is.dim, d)
     npts *= (ie.in_direction(d) - is.in_direction(d)) / 2 + 1;
   ptrdiff_t *index_array = new ptrdiff_t[npts];
+  memset(index_array, 0, sizeof(ptrdiff_t) * npts);
```

Comment:
This doesn't seem like the right fix.  All `idx_vol` points of `index_array` and `amps_array` are initialized.   The problem is in `src_vol::==`.